server.py

```diff
@@ -78,138 +92,165 @@ def room_description(self, room_number):
         :param room_number: int
         :return: str
         """
-
-        # TODO: YOUR CODE HERE
-
-        pass
+        rooms = {
+            1: "You are in the Red Room",
+            2: "You are in the Black Room",
+            3: "You are in the Blue Room",
+            0: "You are in the White Room",
+        }
+        return rooms[room_number]
 
     def greet(self):
         """
         Welcome a client to the game.
-        
+
         Puts a welcome message and the description of the client's current room into
         the output buffer.
-        
-        :return: None 
+
+        :return: None
         """
-        self.output_buffer = "Welcome to {}! {}".format(
-            self.game_name,
-            self.room_description(self.room)
+        self.output_buffer = (
+            f"Welcome to {self.game_name}! {self.room_description(self.room)}"
         )
 
     def get_input(self):
         """
+
         Retrieve input from the client_connection. All messages from the client
         should end in a newline character: '\n'.
-        
+
         This is a BLOCKING call. It should not return until there is some input from
         the client to receive.
-         
-        :return: None 
+
+        :return: None
         """
 
-        # TODO: YOUR CODE HERE
+        received = b""
+        while b"\n" not in received:
+            received += self.client_connection.recv(16)
 
-        pass
+        self.input_buffer = received.decode()
 
     def move(self, argument):
         """
         Moves the client from one room to another.
-        
+
         Examines the argument, which should be one of:
-        
+
         * "north"
         * "south"
         * "east"
         * "west"
-        
+
         "Moves" the client into a new room by adjusting self.room to reflect the
         number of the room that the client has moved into.
-        
+
         Puts the room description (see `self.room_description`) for the new room
         into "self.output_buffer".
-        
+
         :param argument: str
         :return: None
         """
 
-        # TODO: YOUR CODE HERE
-
-        pass
+        directions = {
+            "north": "self.rooms[self.room].north",
+            "south": "self.rooms[self.room].south",
+            "west": "self.rooms[self.room].west",
+            "east": "self.rooms[self.room].east",
+        }
+
+        if eval(directions[argument]) is not None:
+            self.room = eval(directions[argument])
```

Comment:
You should avoid `eval` whenever possible as it allows remote code execution. i.e. security risk.